### PR TITLE
config: Test last assignment wins with defaults

### PR DESCRIPTION
**Pull request by mcassaniti**
@DaanDeMeyer Please use this to test #1836. It's no good relying on the CI side of things yet because it will fail until #1836 lands. If the tests are incorrect then either update the PR directly as required or merge the tests into #1836.

### CODE REVIEW

**Comment by mcassaniti** [comment]
@DaanDeMeyer I've successfully ran the test suite against #1836 and it is passing.

**Review by DaanDeMeyer — CHANGES_REQUESTED** [review]
Can we just write the configuration files from the test itself in a temporary directory? I think that makes all of this a more flexible

**Comment by mcassaniti** [comment]
> Can we just write the configuration files from the test itself in a temporary directory? I think that makes all of this a more flexible

It wouldn't be worth it. We're running most of the tests against a static set of files, then dynamically generating and changing configuration for the last two tests. At best a copy of the static configuration could be done to a temporary directory, but the writes of drop-in files will then likely be in order.

File: tests/test_config.py [hunk 1]
```diff
@@ -1,6 +1,72 @@
...
+
+    configs = list()
+    for i in range(len(paths)):
+        configs.append(parse_paths(paths[:i+1]))
+
+    return configs
+
+
+@contextlib.contextmanager
+def cd_new_dir(path: Path) -> Iterator[None]:
```
Comment (behrmann) [inline tests/test_config.py:50]: Add a todo, this can be dropped once we are on Python 3.11

**Comment by mcassaniti** [comment]
I don't know what is going on with the CentOS disk test but I highly doubt adding a test suite will affect it.

**Comment by behrmann** [comment]
Removing the CI fails label because the CentOS failure is unrelated.

**Comment by behrmann** [comment]
Let's merge this and fix them later, since they have some benefit.

**Pull request merged**